cmd/boulder-wfe2/main.go

```diff
@@ -94,50 +94,51 @@ type config struct {
 // loadCertificateFile loads a PEM certificate from the certFile provided. It
 // validates that the PEM is well-formed with no leftover bytes, and contains
 // only a well-formed X509 certificate. If the cert file meets these
-// requirements the PEM bytes from the file are returned, otherwise an error is
-// returned. If the PEM contents of a certFile do not have a trailing newline
-// one is added.
-func loadCertificateFile(aiaIssuerURL, certFile string) ([]byte, error) {
+// requirements the PEM bytes from the file are returned along with the parsed
+// certificate, otherwise an error is returned. If the PEM contents of
+// a certFile do not have a trailing newline one is added.
+func loadCertificateFile(aiaIssuerURL, certFile string) ([]byte, *x509.Certificate, error) {
```

Comment:
I had to take back my :+1: on this idea. One of the main jobs of `loadCertificateFile` is making sure there is a trailing `\n` in the PEM for the certificate to ensure the chain building later on produces the correct RFC 8555 response chain. Using `cert.Raw` bytes will mean having to move the PEM encoding to the caller, and also handling the trailing newline. Since this function is only used in this one binary and its only one more return argument I think it's better as-written.

Reply:
Ah, interesting. And also I had missed the fact that these `[]byte` being returned were PEM bytes rather than DER bytes, so that also makes my recommendation for `cert.Raw` not work. LGTM.